iRacing/iRacing.cs

```diff
@@ -1531,168 +1530,64 @@ public override void Init(PluginManager pluginManager)
-            jokerLapChecker = false;
-            finishedCars = new List<string> { };
-            fuelTargetCheck = false;
-            oldFuelValue = 0;
-            NBactive = false;
-            NBvalue = false;
-            NBspeedLim = false;
-            ERSlapCounter = 0;
-            ERSstartingLap = true;
-            TCon = false;
-            TCduration = 0;
-            offTrack = false;
-            IRchange = 0;
-
-            //Props that need refresh
-            Base.SetProp("TCActive", false);
-
-            //Refreshing some lists
-            if (Base.counter == 59)
-            {
-                realGapLocks.Clear();
-                realGapChecks.Clear();
-                realGapPoints.Clear();
-                realGapOpponentDelta.Clear();
-                realGapOpponentRelative.Clear();
-                sessionCarsLapsSincePit.Clear();
-                sessionCarsLap.Clear();
-
-                lapDeltaCurrent.Clear();
-                lapDeltaSessionBest.Clear();
-                lapDeltaLast.Clear();
-                lapDeltaRecord.Clear();
-                lapDeltaLastChange.Clear();
-                lapDeltaSessionBestChange.Clear();
-                lapDeltaLapRecordChange.Clear();
-                lastChunks.Clear();
-                SBChunks.Clear();
-                LRChunks.Clear();
-
-                for (int u = 0; u < trackSections; u++)
-                {
-                    List<bool> locks = new List<bool> { };
-                    List<bool> checks = new List<bool> { };
-                    List<TimeSpan> points = new List<TimeSpan> { };
-
-                    for (int i = 0; i < 64; i++)
-                    {
-                        locks.Add(false);
-                        checks.Add(false);
-                        points.Add(TimeSpan.FromSeconds(0));
-                    }
-
-                    realGapLocks.Add(locks);
-                    realGapChecks.Add(checks);
-                    realGapPoints.Add(points);
-                }
-
-                for (int i = 0; i < 64; i++)
-                {
-                    realGapOpponentDelta.Add(0);
-                    realGapOpponentRelative.Add(0);
-                    sessionCarsLapsSincePit.Add(-1);
-                    sessionCarsLap.Add(-1);
-                }
-
-                for (int i = 0; i < lapDeltaSections + 1; i++)
-                {
-                    lapDeltaCurrent.Add(-1);
-                    lapDeltaSessionBest.Add(-1);
-                    lapDeltaLast.Add(-1);
-                    lapDeltaRecord.Add(-1);
-                    lapDeltaLastChange.Add(0);
-                    lapDeltaSessionBestChange.Add(0);
-                    lapDeltaLapRecordChange.Add(0);
-                }
-                for (int i = 0; i < deltaChangeChunks; i++)
-                {
-                    lastChunks.Add(0);
-                    SBChunks.Add(0);
-                    LRChunks.Add(0);
-                }
-            }
-        }
-
```

Comment:
Where did all this go?

Reply:
DataUpdateIdle has been deleted.
We still have the initialization of those lists in the Init() class (see line 479 - 525).

When the game is offline the whole class will be destroyed and re-created when needed. Thus, an explicit initialization (DataUpdateIdle) should not be needed.

Reply:
To be more specific, the magic happens in DataPlugin.cs

I changed the main program to call the DataUpdate() method of all sections only when some game is running. Otherwise they will be cleared and initialized here.

```
            if (data.GameRunning && data.NewData != null)
            {
                dahlGameData.GameData = data;
                dahlGameData.GameName = data.GameName;
                dahlGameData.SetGameData();

                foreach (SectionBase section in Sections)
                {
                    section.DataUpdate();
                }
            } else
            {
                Sections.Clear();
                InitSections();
            }
```

Reply:
Ok, looks like you changed this around 19.00 yesterday? I tested around 21.00 and the dashboard bindings werent working. Should the screen class be added here as well?

Reply:
@andreasdahl1987 Let me check this in the evening. The screen class should not be added here, it's a subclass of Dashboard